src/randomizer.py

```diff
@@ -865,6 +865,186 @@ def minus1(node: ca.Node) -> ca.BinaryOp:
+        """Try moving parentheses to the left side sometimes (sadly, it seems to matter)"""
+        if random.choice([True, False]) and right.op in ['+', '-']:
+            # ((a + b) - c)
+            return ca.BinaryOp(right.op, ca.BinaryOp('+', left, right.left), right.right)
+        else:
+            # (a + (b - c))
+            return ca.BinaryOp('+', left, right)
+
+    # Conversions
+    def to_array(node: ca.BinaryOp) -> ca.ArrayRef:
+        """Change a BinaryOp StructRef (of the form (a + b)->c or (*(a + b)).c)
+        to an ArrayRef StructRef of the form a[b].c"""
+        # Handle different expressions like (a - 1)->c
+        if (node.op == '-'):
+            node.right = ca.UnaryOp('-', node.right)
+        return ca.ArrayRef(node.left, node.right)
+
+    def to_binop(node: ca.ArrayRef) -> ca.BinaryOp:
+        """Change an ArrayRef StructRef (of the form a[b].c or (&(a[b]))->c)
+        to an BinaryOp StructRef of the form (a + b)->c"""
+        if isinstance(node.subscript, ca.BinaryOp):
+            return randomize_associative_binop(node.name, node.subscript)
+        return ca.BinaryOp('+', node.name, node.subscript)
+
+    def deref(node: ca.Node) -> ca.UnaryOp:
+        """Surround the given node with a dereference operator"""
+        # This should be possible but it would mess up my crappy list handling code :|
+        #
+        #if isinstance(node, ca.UnaryOp) and node.op == '&':
+        #    return node.expr
+        return ca.UnaryOp('*', node)
+
+    def addr(node: ca.Node) -> ca.UnaryOp:
+        """Surround the given node with an address-of operator"""
+        # This should be possible but it would mess up my crappy list handling code :|
+        #
+        #if isinstance(node, ca.UnaryOp) and node.op == '*':
+        #    return node.expr
+        return ca.UnaryOp('&', node)
+
+    def rec(node: ca.Node, nodes: List[ca.Node]) -> Tuple[bool, bool]:
+        """Add each child node of the StructRef to a list, stopping when the node is not a
+        dereference or address-of unary operator. Returns 'failed' when a UnaryOp with an
+        op other than * or & was encountered."""
+        if isinstance(node, ca.UnaryOp):
+            if not node.op in ['&', '*']:
+                return False, True
+            else:
+                binop_or_array_ref, failed = rec(node.expr, nodes)
+                if failed:
+                    return binop_or_array_ref, failed
+                nodes.append(node)
+                return binop_or_array_ref, False
+        else:
+            nodes.append(node)
+            return isinstance(node, (ca.ArrayRef, ca.BinaryOp)), False
+
+        return False, True
+
+    # (Oh god why)
+    def apply_child(parent: ca.Node, func) -> None:
+        if isinstance(parent, ca.StructRef):
+            parent.name = func(parent.name)
+        elif isinstance(parent, ca.UnaryOp):
+            parent.expr = func(parent.expr)
+
+    def set_child(parent: ca.Node, child: ca.Node) -> None:
```

Comment:
same as `apply_child(parent, lambda _: child)` (moving this comment up here; I accidentally commented on the test before)